recovery/elliptics_recovery/types/dc.py

```diff
@@ -222,23 +227,71 @@ def contains(addresses_with_backends, address, backend_id):
     return address_range
 
 
-def final_merge(ctx, results):
-    import shutil
-    ctx.stats.timer('main', 'final_merge')
-    log.info("final merge")
-
-    ctx.merged_filename = os.path.join(ctx.tmp_dir, 'merged_result')
-    dump_filename = os.path.join(ctx.tmp_dir, 'dump')
-    with open(ctx.merged_filename, 'wb') as mf:
-        with open(dump_filename, 'wb') as df:
-            for res in (r for r in results if r):
-                shutil.copyfileobj(open(res[0], 'rb'), mf)
-                os.remove(res[0])
-                shutil.copyfileobj(open(res[1], 'rb'), df)
-                os.remove(res[1])
-
-    log.debug("Merged_filename: %s, address: %s, groups: %s, tmp_dir: %s",
-              ctx.merged_filename, ctx.address, ctx.groups, ctx.tmp_dir)
+def dump_key(ctx, rest_file, key, key_infos, newest_key_group):
+    if key_infos[0].group_id != newest_key_group:
+        index = [k.group_id for k in key_infos].index(newest_key_group)
+        tmp = list(key_infos)
+        tmp[0], tmp[index] = tmp[index], tmp[0]
+        key_infos = tuple(tmp)
```

Comment:
First key_info in key_infos list must correspond to largest bucket among other buckets. `newest_key_group` is a group_id of the largest bucket.